portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,21 +26,40 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> greetings;
+  private List<String> comments = new ArrayList<>();
+  private Gson gson = new Gson();
 
   @Override
-  public void init() {
-    greetings = new ArrayList<>();
-    greetings.add("Hello, Carolyn!");
-    greetings.add("Khairei, Carolyn!");
-    greetings.add("Privyet, Carolyn!");
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String json = gson.toJson(comments);
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(greetings);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+
+    // Respond with the result.
+    String json = gson.toJson(text);
+    comments.add(json);
+
+    // Redirect back to the proper container.
     response.setContentType("text/html;");
     response.getWriter().println(json);
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
Dawn told me to use an optional for this function. 

Optional<String> getParameter(HttpServletRequest request, String param)
Effective Java Item 55: Return optionals judiciously

Reply:
Thanks for letting me know!